src/npc.cpp

```diff
@@ -620,6 +620,13 @@ void npc::revert_after_activity()
 {
     mission = previous_mission;
     attitude = previous_attitude;
+    activity = player_activity();
+    current_activity.clear();
+    clear_destination();
+    // Clear any backlog items
+    for( auto backlog_item = backlog.begin(); backlog_item != backlog.end(); ) {
+        backlog_item = backlog.erase( backlog_item );
```

Comment:
```suggestion
    backlog.clear();
```
As replacement for the whole loop and the comment.